test/e2e/storage/framework/testdriver.go

```diff
@@ -156,6 +156,7 @@ const (
 	CapExec               Capability = "exec"               // exec a file in the volume
 	CapSnapshotDataSource Capability = "snapshotDataSource" // support populate data from snapshot
 	CapPVCDataSource      Capability = "pvcDataSource"      // support populate data from pvc
+	CapAttachRequired     Capability = "attachRequired"     // attachRequired indicates this CSI volume driver requires an attach operation (because it implements the CSI ControllerPublishVolume())
```

Comment:
There may be some volume types that support attach but can support snapshotting while attaching/detaching. So maybe this capability needs to be more specific like "offlineSnapshotClone"